ethereum_history_api/oracles/src/noir/circuit/nargoProver.ts

```diff
@@ -0,0 +1,65 @@
+import path from 'path';
+import { MonorepoCircuit } from './circuit.js';
+import { $ } from 'execa';
+import toml from '@iarna/toml';
+import { InputMap } from '@noir-lang/noirc_abi';
+import { readFile, unlink, writeFile } from 'fs/promises';
+import { readTomlObject } from '../../util/file.js';
+import { addHexPrefix } from '../../util/hex.js';
+import { type Hex } from 'viem';
+
+export interface NargoProve {
+  proof: Hex;
+  verifierData: InputMap;
+}
+
+// IMPORTANT: The proof paths used here are not unique to the `proofId` - therefore they can be overridden in parallel proof generation.
+// https://github.com/noir-lang/noir/issues/5037
+export class NargoProver {
+  constructor(
+    public circuit: MonorepoCircuit,
+    public proofId: string
+  ) {}
+
+  public proverName(): string {
+    return `Prover_${this.proofId}`;
+  }
+
+  public proverTomlPath(): string {
+    return path.join(this.circuit.packagePath(), `${this.proverName()}.toml`);
+  }
+
+  public verifierName(): string {
+    return `Verifier_${this.proofId}`;
+  }
+
+  public verifierTomlPath(): string {
+    return path.join(this.circuit.packagePath(), `${this.verifierName()}.toml`);
+  }
+
+  public proofPath(): string {
+    return path.join(this.circuit.root, 'proofs', `${this.circuit.name}.proof`);
+  }
+
+  public proofAsFieldsPath(): string {
+    return path.join(this.circuit.root, 'proofs', `${this.circuit.name}.proof.json`);
+  }
+
+  public async proveInternal(): Promise<void> {
```

Comment:
Maybe this method should be private

Reply:
I aslo seen notation `doProve` for internal core part of a method, but in this case I would call it `executeProveCommand`